File: UI/wwwroot/Model/Secciones.js
//@ts-check
// @ts-ignore
import { EntityClass } from '../WDevCore/WModules/EntityClass.js';
import { Docente_materias } from './Docente_materias.js';
import { Estudiante_clases } from './Estudiante_clases.js';
import { Evaluaciones } from './Evaluaciones.js';
class Secciones extends EntityClass {
   /** @param {Partial<Secciones>} [props] */
   constructor(props) {
       super(props, 'EntityDbo');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id;
   /**@type {String}*/ Nombre;
   /**@type {Number}*/ Clase_id;
   /**@type {Number}*/ Docente_id;
   /**@type {String}*/ Observaciones;
   /**@type {Date}*/ Created_at;
   /**@type {Date}*/ Updated_at;
   /**@type {Array<Docente_materias>} OneToMany*/ Docente_materias;
   /**@type {Array<Estudiante_clases>} OneToMany*/ Estudiante_clases;
   /**@type {Array<Evaluaciones>} OneToMany*/ Evaluaciones;
}
export { Secciones };

File: AppGenerateFiles/dbo/Model/Secciones.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Secciones : EntityClass {
       [PrimaryKey(Identity = false)]
       public int? Id { get; set; }
       public string? Nombre { get; set; }
       public int? Clase_id { get; set; }
       public int? Docente_id { get; set; }
       public string? Observaciones { get; set; }
       public DateTime? Created_at { get; set; }
       public DateTime? Updated_at { get; set; }
       [OneToMany(TableName = "Docente_materias", KeyColumn = "Id", ForeignKeyColumn = "Seccion_id")]
       public List<Docente_materias>? Docente_materias { get; set; }
       [OneToMany(TableName = "Estudiante_clases", KeyColumn = "Id", ForeignKeyColumn = "Seccion_id")]
       public List<Estudiante_clases>? Estudiante_clases { get; set; }
       [OneToMany(TableName = "Evaluaciones", KeyColumn = "Id", ForeignKeyColumn = "Seccion_id")]
       public List<Evaluaciones>? Evaluaciones { get; set; }
   }
}
